librz/asm/arch/rx/rx.c

```diff
@@ -0,0 +1,163 @@
+
+	if (opr->kind == RX_OPERAND_FLAG) {
+		rz_strf(buf->buf, "%s", RxNameFlag(opr->v.flag));
+		return true;
+	}
+
+	if (opr->kind == RX_OPERAND_IMM) {
+		rz_strf(buf->buf, "#IMM:%d(#0x%" PFMT32x ")",
+			opr->v.imm.imm_width,
+			opr->v.imm.imm);
+		return true;
+	}
+
+	if (opr->kind == RX_OPERAND_REG) {
+		if (opr->v.reg.dsp_width) {
+			// dsp value
+			rz_strbuf_appendf(buf, "dsp:%d(#0x%" PFMT32x ")",
+				opr->v.reg.dsp_width,
+				opr->v.reg.dsp_val);
+		}
+
+		if (opr->v.reg.as_indirect) {
+			rz_strbuf_appendf(buf, "[");
+		}
+		if (opr->v.reg.fix_mode == RX_FIXOP_PRE_DEC) {
+			rz_strbuf_appendf(buf, "-");
+		}
+
+		if (opr->v.reg.as_base) {
+			rz_strbuf_appendf(buf, "%s,%s",
+				RxNameReg(opr->v.reg.ri),
+				RxNameReg(opr->v.reg.reg));
+		} else {
+			rz_strbuf_appendf(buf, "%s", RxNameReg(opr->v.reg.reg));
+		}
+
+		if (opr->v.reg.fix_mode == RX_FIXOP_POST_INC) {
+			rz_strbuf_appendf(buf, "+");
+		}
+		if (opr->v.reg.as_indirect) {
+			rz_strbuf_appendf(buf, "]");
+		}
+
+		if (opr->v.reg.memex != RX_EXT_NON) {
+			rz_strbuf_appendf(buf, ".%s", RxNameExt(opr->v.reg.memex));
+		}
+	}
+
+	return true;
+}
+
+bool rx_inst_stringify(RxInst *inst, RzStrBuf *buf) {
+	// construct string output to RzStrBuf
+	if (inst->op == RX_OP_INVALID) {
+		return false;
+	}
+
+	RzStrBuf opr0_buf, opr1_buf, opr2_buf;
+	rz_strbuf_init(&opr0_buf);
+	rz_strbuf_init(&opr1_buf);
+	rz_strbuf_init(&opr2_buf);
+
+	bool has_opr0 = inst->v0.kind != RX_OPERAND_NULL &&
+		rx_operand_stringify(inst, &inst->v0, &opr0_buf);
+	bool has_opr1 = inst->v1.kind != RX_OPERAND_NULL &&
+		rx_operand_stringify(inst, &inst->v1, &opr1_buf);
+	bool has_opr2 = inst->v2.kind != RX_OPERAND_NULL &&
+		rx_operand_stringify(inst, &inst->v2, &opr2_buf);
+
+	if (inst->op == RX_OP_BCND_W || inst->op == RX_OP_BCND_B || inst->op == RX_OP_BCND_S ||
+		inst->op == RX_OP_BMCND || inst->op == RX_OP_SCCOND) {
+		if (has_opr0 && inst->v0.kind == RX_OPERAND_COND) {
+			// build b[cnd]
+			RzStrBuf cond_buf;
+			rz_strbuf_init(&cond_buf);
+			rz_strf(cond_buf.buf, "%s", RxNameCond(inst->v0.v.cond.cond));
+			rz_strbuf_appendf(buf, RxNameOp(inst->op), cond_buf.buf);
+
+			if (!inst->v0.v.cond.pc_dsp_len) {
+				has_opr0 = false;
+			}
+		} else {
+			rz_strbuf_appendf(buf, "%s[invalid]",
+				RxNameOp(inst->op));
+		}
+	} else {
+		rz_strbuf_appendf(buf, "%s", RxNameOp(inst->op));
+	}
+
+	if (inst->sz_mark != RX_EXT_NON) {
+		rz_strbuf_appendf(buf, ".%s ", RxNameExt(inst->sz_mark));
+	} else {
+		rz_strbuf_appendf(buf, " ");
+	}
+
+	if (has_opr0) {
+		rz_strbuf_appendf(buf, "%s, ", opr0_buf.buf);
+	}
+	if (has_opr1) {
+		rz_strbuf_appendf(buf, "%s, ", opr1_buf.buf);
+	}
+	if (has_opr2) {
+		rz_strbuf_appendf(buf, "%s", opr2_buf.buf);
+	}
+
+	return true;
+}
+
+/**
+ * Parse binary data to RxInst according to RxDesc
+ * \param inst
+ * \param bytes_read
+ * \param buf
+ * \param buf_len
+ * \return
+ */
+RZ_API bool rx_dis(RxInst RZ_OUT *inst, st32 RZ_OUT *bytes_read, const ut8 *buf, size_t buf_len) {
```

Comment:
Add `rz_return..` assert and `RZ_NONNULL` for `inst` and `bytes_read`, maybe `buf`?

Reply:
marked, will fix it